src/net.h

```diff
@@ -861,6 +861,12 @@ class CNode
         return m_conn_type == ConnectionType::INBOUND;
     }
 
+    /* Whether we send addr messages over this connection */
+    bool RelayAddrsWithConn() const
```

Comment:
nit: not a big fan of this name... I actually think `IsAddrRelayPeer` is much better :)
I already saw some discussion around the distinction between "node" and "peer", and now you start referring to it as a "connection".

Reply:
Also, this method should be now used here: https://github.com/bitcoin/bitcoin/blob/4c66cc0a5a99cbad51448e51d6b28e11e5232e33/src/net.cpp#L2772

Reply:
I think the new code should be:

```
    if (conn_type_in != ConnectionType::BLOCK_RELAY) {
        m_tx_relay = MakeUnique<TxRelay>();
    }
    if (IsAddrRelayPeer()) {
        m_addr_known = MakeUnique<CRollingBloomFilter>(5000, 0.001);
    }
```

Reply:
ah nice. I like it! I was also concerned about the duplicated logic making room for future discrepancy. I've updated to include this, thanks! 

RE name: seems like an issue with the whole convention around connection types? The name I've chosen is consistent with all the `m_conn_type` functions.